### PR TITLE
Added foxx executor and minor fixes for grequest session.

### PR DESCRIPTION
**Pull request by dilipvamsi**
Class uses the connection's session for making a call to the foxx function.

### CODE REVIEW

**Comment by dothebart** [comment]
hm, having tests for this seems hard? maybe one could use the default aardvark routes for some tests?

**Comment by dilipvamsi** [comment]
Hi @dothebart ,
Should the functions take `mount` as an argument
```python
def get(self, mount, url, **kwargs)
```
Or
```python
def get(self, url, **kwargs)
```
Which one is better?

**Comment by dothebart** [comment]
I'd use `mountPoint` as thats what i.e. aardvark uses as strings

**Comment by dilipvamsi** [comment]
@dothebart , I have added functionality to call from a service or directly with url.
Review for any more changes that are needed to be done.

File: pyArango/tests/tests.py [hunk 1]
```diff
@@ -852,6 +852,14 @@ def test_users_update(self) :
 
         Connection(arangoURL=ARANGODB_URL, username="pyArangoTest_tesla", password="newpass")
 
+    def test_foxx(self):
```
Comment (dothebart) [inline pyArango/tests/tests.py]: hm, shouldn't this be rather something other than fetching a html file? We meanwhile did a good job in migrating away most of aardvarks routes to native arangodb routes ;-) however there still seems to be i.e.: 
`POST /_db/_system/_admin/aardvark/graph-examples/create/mps_graph`
which probably will remain available and do something where you can revalidate that it worked - plus a parseable json reply. 

`/_db/_system/` should come from the selected db this foxx is working on. `_admin/aardvark` is the mountPoint, `create` the actual route, `mps_graph` a URL parameter.
if you put in a wrong graph, it throws an error, which you should convert to a python exception (imho?) 

I'd rather see the use of it a bit like that: you derive the foxx base class, implement the routes to query. You'll then create an instance of it by specifying its mountpoint. 
In optimal, the client stub could be generated from a swagger json, but by hand should be ok. So here: 
```
class aardvarkTest(_BaseFoxx):
      def __init__(self, db, mountPoint):....
      def graph_examples_create(self):
...
```
and then the instance: 
```
adminAardvark = db.getFoxx(aardvarkTest, "_admin/aardvark")
```

Comment (dilipvamsi) [reply pyArango/tests/tests.py]: Hi @dothebart ,
Can you direct me to the page which has the aardvarks routes?
So, I can add some examples.

**Comment by tariqdaouda** [comment]
Hi,

Thanks for this, could you please resend it to the Dev branch (after fixing the aardvark).

**Comment by dilipvamsi** [comment]
> Hi,
> 
> Thanks for this, could you please resend it to the Dev branch (after fixing the aardvark).

okay

**Comment by dothebart** [comment]
hint: you can edit the target branch along with the subject of the PR - not all that intuitive ;-)

**Comment by dothebart** [comment]
superseeded by: https://github.com/ArangoDB-Community/pyArango/pull/149

**Pull request closed without merging**